### PR TITLE
Add support for Daisy Seed rev5

### PR DESCRIPTION
**Pull request by miselaytes-anton**
# Context
Daisy Seed 1.1 [uses a different audio codec](https://github.com/electro-smith/libDaisy/pull/402) (WM8731 instead of AK4556).

# Notes

Unfortunately, I can not make it produce the sound when I run the `passthru` example. I suspect it might be because of the change below, but I am not sure how to change the direction in audio.rs.
```cpp
sai_config.a_dir           = SaiHandle::Config::Direction::TRANSMIT;
sai_config.b_dir           = SaiHandle::Config::Direction::RECEIVE;
```
changing to:

```cpp
sai_config.a_dir           = SaiHandle::Config::Direction::RECEIVE;
sai_config.b_dir           = SaiHandle::Config::Direction::TRANSMIT;
```
see [here](
https://github.com/electro-smith/libDaisy/pull/402/files)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -28,6 +28,7 @@ cortex-m-log = { version = "~0.7", features = ["itm", "semihosting", "log-integr
 panic-itm = { version = "~0.4.2", optional = true  }
 panic-semihosting = { version = "0.5.6", optional = true  }
 cortex-m-semihosting = { version = "0.3.7", optional = true  }
+wm8731-alt = { git = "https://github.com/YruamaLairba/wm8731-alt", branch = "main" }
```
Comment (miselaytes-anton) [inline Cargo.toml:31]: I [asked the author](https://github.com/YruamaLairba/wm8731-alt/issues/1) if he could publish it to crates.io. Alternatively we can use a [different crate](https://docs.rs/wm8731/latest/wm8731/).

File: src/codec_wm8731.rs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        .iwl()
+        .iwl_24_bits()
+        .ms()
+        .slave()
+        .into_command();
+    wm8731.send(cmd);
+    delay_ms(10);
+
+    // Sample rate.
+    // For a 12.288 MHz master clock.
```
Comment (miselaytes-anton) [inline src/codec_wm8731.rs:116]: Not sure if this is correct value. Alternatives are  is 18.432 , 11.2896, 16.9344, 12MHz

File: Cargo.toml (hunk 1 above)
Comment (mtthw-meyer) [reply Cargo.toml:31]: We can merge it like this for now. Can change it later as needed.

**Comment by mtthw-meyer** [comment]
Did you get the sound working?

**Comment by miselaytes-anton** [comment]
@mtthw-meyer
> Did you get the sound working?

Short answer is no.

Long answer is if I enable bypass from line in to line out then I hear audio. But if I disable bypass, then I do not. Based on that I assume that the codec is configured correctly. 

I was also able to change SAI master direction from `transmit` to `receive` and it no longer hangs. So looks like SAI is also working:
```
22:31:09.825 INFO - Starting system init 
22:31:10.010 INFO - Setting up GPIOs...
22:31:10.010 INFO - Setting up SDRAM...
22:31:10.010 INFO - SDRAM Memory Size 0x19
22:31:10.010 INFO - Setup up Audio...
22:31:10.010 INFO - Setup up DMA...
22:31:10.010 INFO - Memory Size 0xe
22:31:10.010 INFO - Setup up SAI...
22:31:10.010 INFO - Sai1 fifo waiting to receive data.
22:31:10.010 INFO - Audio started!
22:31:10.010 INFO - Setup WM8731 audio codec...
22:31:10.058 INFO - 0x30000300, 0x30000000
22:31:10.058 INFO - System init done!
22:31:10.058 INFO - Startup done!
```


So my remaining suspect is DMA config. I checked SAI register values and I see that I get `overrun` error on SAI_ADR register, so looks like audio data arrives from codec but is not picked up by DMA.
I also see `underrun` error on SAI_BDR register, so DMA never transmits data there. 

Screenshot 2022-02-18 at 22 42 17

**Comment by miselaytes-anton** [comment]
I can not make this work unfortunately.

**Pull request closed without merging**